server/segment_rpc.go

```diff
@@ -58,10 +58,24 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// oInfo will be non-nil if we need to send an updated net.OrchestratorInfo
+	// to the broadcaster
+	var oInfo *net.OrchestratorInfo
+
 	if err := orch.ProcessPayment(payment, segData.ManifestID); err != nil {
 		glog.Errorf("Error processing payment: %v", err)
-		http.Error(w, err.Error(), http.StatusPaymentRequired)
-		return
+
+		if !acceptablePaymentError(err) {
+			http.Error(w, err.Error(), http.StatusPaymentRequired)
+			return
+		}
+
+		oInfo, err = orchestratorInfo(orch, getPaymentSender(payment), orch.ServiceURI().String(), segData.ManifestID)
```

Comment:
Hm upon further inspection it seems that the current orchestrator behavior is to generate a fresh random manifestID as the storage prefix for its OS session every time a broadcaster calls `GetOrchestrator()` on the orchestrator. So, in order to preserve this behavior, we should also generate a fresh random manifestID here instead of using `segData.ManifestID` which is the broadcaster supplied manifestID.

Reply:
Updated to always [generate a random manifestID for the orchestrator OS session](https://github.com/livepeer/go-livepeer/pull/905/files#diff-44bb03d888e11d6429223a7b0e55220dR220).

Reply:
I can't think of a reason why generating a new OS session mid-stream might cause issues either. That looks like a good solution to me!